src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -396,19 +397,15 @@ public void execute(BuildScanExtension buildScan) {
             }
 
             if (isNotEmpty(gitRepo) && isNotEmpty(gitCommitId)) {
-                if (gitRepo.contains("github.com/") || gitRepo.contains("github.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)github\\.com[/|:](.*)").matcher(gitRepo);
-                    if (matcher.matches()) {
-                        String rawRepoPath = matcher.group(2);
-                        String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
-                        buildScan.link("Github source", "https://github.com/" + repoPath + "/tree/" + gitCommitId);
-                    }
-                } else if (gitRepo.contains("gitlab.com/") || gitRepo.contains("gitlab.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)gitlab\\.com[/|:](.*)").matcher(gitRepo);
-                    if (matcher.matches()) {
-                        String rawRepoPath = matcher.group(2);
-                        String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
-                        buildScan.link("GitLab Source", "https://gitlab.com/" + repoPath + "/-/commit/" + gitCommitId);
+                Matcher matcher = Pattern.compile("(https://|git@)(.+?)[:|/](.*)").matcher(gitRepo);
```

Comment:
Shouldn't that be limited to `Github` / `Gitlab`?
Not sure if with this approach we could catch other cases as well (_Bitbucket_...)?
Not a big deal as you are verifying afterward, but I would avoid extra computation if not required.

Reply:
`GITHUB_SERVER_URL` will only be expected to be set if executing on a GitHub Actions runner though, right?

Reply:
CLosing this thread since it has been clarified in the spec.